### ISSUE TITLE
Add the list of recommended add-ons

### ISSUE THREAD

**Issue by amakriLexa04** [wesnoth/wesnoth#9396]
### Describe the desired feature

There are many UMCs that have good gameplay, expand on the half-empty lore of Irdya, and deserve to be considered canon. However, they often don't qualify to be added to the mainline because of their mechanics, high complexity or any other variety of reasons. And that's **not cool**.

That's why I propose that Wesnoth's developers, such as Master of SP Content(@Dalas121) and Project Manager(@Pentarctagon), form a council to review various add-ons and determine whether their story can be considered canon.

If not, then understandable, have a nice day. But if it is, then such campaigns should be honored and shared with people. But adding everything to the mainline would be unreasonable, because it will increase the size of the game's files, so I propose a solution. 

**Add a button to the campaign menu where the user can download/uninstall only those add-ons that have been approved by the developers.** 

I'm attaching 2 simple mockups that I drew in 1 minute in Paint to make it clearer how I imagine it.

I would also like to emphasize that **this will significantly expand the official lore of Irdya without burdening the developers with additional work.**

![Знімок екрана (960)](https://github.com/user-attachments/assets/1ff20879-9c15-4f44-9844-c372a4a4b07c)
![Знімок екрана (961)](https://github.com/user-attachments/assets/7d774e46-4b5b-4ea3-a123-880ede2c39c6)

**Comment by soliton-** [comment]
It's a bit mind boggling how you consider reviewing all addons now and presumably forever in the future as no additional work.

**Comment by amakriLexa04** [comment]
I may have misspoken, but I meant that the developers review only those add-ons whose authors have requested the review.

**Comment by soliton-** [comment]
I'd much rather we implement #8478 before thinking about reviewing/rating addons and creating the potential for lots more work and controversy.

**Comment by amakriLexa04** [comment]
Something like this? If so, I think it indeed will be better.
![Знімок екрана (962)](https://github.com/user-attachments/assets/2a49c1db-7fb6-497b-834c-f8f0a881954b)

**Comment by soliton-** [comment]
Yeah, something like that.

**Comment by Wedge009** [comment]
I could be wrong, but I feel like the admission into the official campaigns collection inherently implies that a UMC campaign is considered canonical. I suppose it's possible that a campaign may not be included for game-play, lack-of-polish, or other reasons, while its plot aspect may still stand in its own right. But for the most part, I think non-inclusion also implies non-canonicity.

I definitely agree with #8478 taking precedence over this in terms of priority.

**Comment by Gothyoba** [comment]
Beyond “canon” add-ons, some kind of official add-on system might be nice. If WML Guide was made official, there woukd be more attention to it, and more people might have an easier time wih WML. Perhaps a soltution to endless copying of music tracks between add-ons could be making the UMC Music Books more easily known by making them “official” in some way.

When it comes to being canon, if an add-on is really so high quality, and has a story that is so inline with mainline, it feels like it should be mainlined. If it shouldn’t be mainlined, then it feels like there would be an issue. There isn’t much of s reason for an imbetween area except for maybe campaigns that will be mainline in the future (e.g. TDG, where this system might lead to more playtesters).

Given how few campaigns enter mainline, I don’t think there is any significant short-term risk of running out of space with campaigns, unless Wensoth’s policies to adding them change significantly.

Maybe this could work a few stories like AOI, but that is low-auality, hence why it’s not in mainline. I personally disagree with the choice of remvojng it, but I, and seemingly the devs as well, agree it is not a high-quality campaign. So while it could be made canon, I don’t see much reason to do it if it right now for a bad campaign.

i don’t think this would be a bad idea, but I think maybe it should mark good, reccomended add-ons instead of canon ones. That also means that one can mark an add-on (e.g. Saving Elensefar) as recommended without dealing with the implications of it being canon.

I suppose for eras which don’t add much to canon this can be considered as long as it is good in other aspects. That way eras that are good but not in mainline style can be considered in this intermediate stage of being a recommended add-on (e.g. War of Legends, Reign of the Lords).

**Comment by soliton-** [comment]
Instead of dumping the work on existing devs a more sustainable approach would be if interested people would form a committee or whatever that works out criteria by which to judge addons and then commit to checking addons regularly to review them. Or require the addon author to request a review or so.

If that is in place then it shouldn't be much work to add support for additional attributes/filters to the addon client to show and filter for recommended addons. How to add the attributes on the server side is a bit more difficult question...

**Comment by Jonathan-Kelly** [comment]
> a more sustainable approach would be if interested people would form a committee or whatever that works out criteria by which to judge addons and then commit to checking addons regularly to review them.

> If that is in place then it shouldn't be much work to add support for additional attributes/filters to the addon client to show and filter for recommended addons. 

A centralized art committee would still likely attract a ton of controversy.

It would be fairer if players could search for content recommended by a particular registered user account - "campaigns recommended by amakriLexa04" or "eras recommended by soliton-". This would allow a decentralized community of trusted content reviewers to arise organically.

Maybe instead of simple recommendations, registered users could apply specific tags like "Canonical Story" and "Innovative Gameplay" to others' content.

**Comment by amakriLexa04** [comment]
Well, your comments have made me realize that my initial proposal was rather bad thought out and impossible to implement in a way that would not burden developers and cause controversy among players. Indeed, to review an add-on properly, you need to spend several hour of your time, research whether its story contradicts the existing lore, review the code and do much more.

The idea of making the campaign menu display a link to the add-on server as in version 1.0 is possible, but we need to go into campaign_selection.cpp and campaign_selection.hpp and do some manipulations there, making something similar to the “Missing Campaigns” message. It's not hard to do, but does it really make sense if you can just rename the “Add-ons” button to something more understandable for new players who have never even heard such word before? 

Anyway, now I have a new idea, inspired by the comments from Jonathan-Kelly and Gothyoba — what if we add support for ratings to the add-on server? The player who has downloaded the add-on will be able to rate this add-on and add the tags they think are necessary. This system is completely controlled by the players, although we should probably take into account that at least some kind of moderation will be needed to prevent players from breaking the system and attaching random tags.

Here's another mockup made in Paint that shows how I see this idea.
![Знімок екрана (970)](https://github.com/user-attachments/assets/f1daed1f-f973-46d1-acdd-a8f6f00b7334)

**Comment by Dalas121** [comment]
I agree that some kind of rating system would be very useful - both for add-ons and mainline. Ideally players could rate individual scenarios as well as overall campaigns, and ratings could be carried over from previous versions as the game updates.

But from a technical perspective, I'm unsure how we could implement a reviews system that's not vulnerable to being exploited. Players don't need accounts/credentials to download from the add-on server, nor to play mainline campaigns. How could such a system be designed to prevent one person from flooding a single campaign/addon with many reviews?

Regarding the original suggestion - it wouldn't be impossible for me to hand-curate a list of addons, but it would definitely be a lot of work, particularly if the list is expected to change often.

**Comment by Gothyoba** [comment]
Inwould personally presume this would be done throuh a forums account. You wouldn’t need a forums acocunt to dowload them or play the game, but you would for rating the add-ons. I would generslly support the initial proposalif it were feasable to do it and it wouldn’t deract from mainline development.
